polkadot/runtime/parachains/src/paras_inherent/mod.rs

```diff
@@ -542,20 +539,25 @@ impl<T: Config> Pallet<T> {
 			.map(|(_session, candidate)| candidate)
 			.collect::<BTreeSet<CandidateHash>>();
 
-		let freed_disputed: BTreeMap<CoreIndex, FreedReason> =
-			<inclusion::Pallet<T>>::collect_disputed(&current_concluded_invalid_disputes)
-				.into_iter()
-				.map(|core| (core, FreedReason::Concluded))
-				.collect();
+		let (freed_disputed_cores, freed_disputed_candidates): (
+			BTreeMap<CoreIndex, FreedReason>,
+			BTreeSet<CandidateHash>,
+		) = <inclusion::Pallet<T>>::collect_disputed(&current_concluded_invalid_disputes)
+			.map(|(core, candidate)| ((core, FreedReason::Concluded), candidate))
+			.unzip();
 
 		// Create a bit index from the set of core indices where each index corresponds to
 		// a core index that was freed due to a dispute.
 		//
 		// I.e. 010100 would indicate, the candidates on Core 1 and 3 would be disputed.
-		let disputed_bitfield = create_disputed_bitfield(expected_bits, freed_disputed.keys());
+		let disputed_bitfield =
+			create_disputed_bitfield(expected_bits, freed_disputed_cores.keys());
 
-		if !freed_disputed.is_empty() {
-			<scheduler::Pallet<T>>::free_cores_and_fill_claimqueue(freed_disputed.clone(), now);
+		if !freed_disputed_cores.is_empty() {
```

Comment:
I think the reason why this was not done yet is because the disputed candidates are further being used for sanitizing the bitfields and candidates, so it's useful to have them returned separately.

However, refactored the code so that we merge the two collections into one and only call `free_cores_and_fill_claimqueue` once